src/components/NavBar/index.js

```diff
@@ -83,42 +103,90 @@ const QAButton = ({ intl, changeDropDownDirection }) => (
 const NavBarRightButtonContainer = styled.div`
   height: 70px;
   line-height: 70px;
-  position: absolute;
-  right: 70px;
-  top: 0px;
+  text-align: center;
+  color: white;
   padding-left: 20px;
   padding-right: 20px;
   border-left: 1px solid rgba(0,0,0,0.2);
+  display: flex;
 `;
 const NavBarRightButton = styled.div`
   height: 30px;
   margin: 20px auto;
   line-height: 30px;
-  text-align: center;
-  color: white;
+  display: flex;
 `;
+const SearchBarFont = styled.div`
+  color: rgba(255, 255, 255, 0.65);
+  padding-left: 14px;
+`;
+const DivSearchBarField = styled.div`
+  margin: auto;
+  display: flex;
+  width: 60%;
+`;
+
+const SearchBarField = ({ intl, classes, store: { search }, onClose, onSearchBarKeyDown }) => (
+  <DivSearchBarField>
+    <div className={`icon iconfont icon-ic_search ${classes.searchBarLeftIcon}`} />
+    <TextField
+      placeholder={intl.formatMessage({ id: 'search.placeholder', defaultMessage: 'Type to begin search' })}
+      className={classes.searchBarTextField}
+      InputProps={{
+        autoFocus: true,
+        disableUnderline: true,
+        classes: {
+          input: classes.searchBarInput,
+        },
+        onKeyDown: (e) => onSearchBarKeyDown(e),
+        onChange: e => {
+          search.phrase = e.target.value;
+          search.loading = true;
+          _.debounce(search.init, 1500)();
+        },
+        value: search.phrase,
+        inputProps: {
+          id: 'searchEventInput',
+        },
+      }}
+    >
+    </TextField>
+    <div className="icon iconfont icon-ic_close" onClick={onClose} />
+  </DivSearchBarField>
+);
 
 const MyActivities = observer(({ store: { global } }) => {
   if (global.userData.totalCount > 0) {
-    return (<NavBarRightButtonContainer>
-      <NavLink to={Routes.ACTIVITY_HISTORY}>
+    return (<NavLink to={Routes.ACTIVITY_HISTORY}>
+      <NavBarRightButtonContainer>
         <NavBarRightButton>
           <Badge badgeContent={global.userData.totalCount} color="secondary">
             <FormattedMessage id="navBar.activities" defaultMessage="My Activities" />
           </Badge>
         </NavBarRightButton>
-      </NavLink>
-    </NavBarRightButtonContainer>);
+      </NavBarRightButtonContainer>
+    </NavLink>);
   }
-  return (<NavBarRightButtonContainer>
-    <NavLink to={Routes.ACTIVITY_HISTORY}>
+  return (<NavLink to={Routes.ACTIVITY_HISTORY}>
+    <NavBarRightButtonContainer>
       <NavBarRightButton>
         <FormattedMessage id="navBar.activities" defaultMessage="My Activities" />
       </NavBarRightButton>
-    </NavLink>
-  </NavBarRightButtonContainer>);
+    </NavBarRightButtonContainer>
+  </NavLink>);
 });
 
+const SearchButton = observer(({ onClick }) => (
```

Comment:
don't pass the `onClick` functionality. call the store func directly from the component. you can `inject('store')` to stateless components.